src/os/source/vectorsource.js

```diff
@@ -3344,7 +3344,7 @@ os.source.Vector.prototype.selectAll = function() {
  * @inheritDoc
  */
 os.source.Vector.prototype.selectNone = function() {
-  var i = this.selected_.length;
+  var i = (this.selected_) ? this.selected_.length : 0;
```

Comment:
This variable is declared as:
```
/**
   * @type {!Array<!ol.Feature>}
   * @private
   */
  this.selected_ = [];
```

Which means it should never be null or undefined. If that's happening, something else is going wrong.

Reply:
Removed!